lib/posix-sysinfo/sysinfo.c

```diff
@@ -72,16 +72,33 @@ static struct utsname utsname = {
 
 UK_SYSCALL_R_DEFINE(int, sysinfo, struct sysinfo *, info)
 {
+#ifdef CONFIG_HAVE_PAGING
+	struct uk_pagetable *pt;
+	__sz total_memory;
+	unsigned int mem_unit = 1;
+#endif /* CONFIG_HAVE_PAGING */
+
 	if (!info)
 		return -EFAULT;
 
 	memset(info, 0, sizeof(*info));
 
 	info->procs = 1; /* number of processes */
 
-#if CONFIG_LIBUKALLOC
-	/* TODO: memory usage can be obtained from ukallloc */
-#endif
+#ifdef CONFIG_HAVE_PAGING
+	pt = ukplat_pt_get_active();
+
+	total_memory = pt->fa->total_memory;
+	while (total_memory > __UL_MAX) {
+		total_memory >>= 1;
+		mem_unit <<= 1;
+	}
+
+	info->totalram = (unsigned long) (pt->fa->total_memory / mem_unit);
```

Comment:
Isn't `pt->fa->total_memory / mem_unit` just `total_memory`?

Reply:
Sometimes. For x86_64 and arm64 (and basically all archs where sizeof(long) == 8) that is true, but you never know what other architectures we might support one day.